subprojects/dependency-management/src/main/java/org/gradle/internal/resolve/ModuleVersionNotFoundException.java

```diff
@@ -128,29 +139,44 @@ private static void addLocations(TreeFormatter builder, Collection<String> locat
         }
         builder.node("Searched in the following locations");
         builder.startChildren();
-        for (String location : locations) {
-            builder.node(location);
-        }
+
+        locations.forEach(builder::node);
+
         builder.endChildren();
-        tryHintAboutArtifactSource(builder, locations);
     }
 
     // This method should ideally use more data to figure out if the message should be displayed
     // or not. In particular, the ivy patterns can make it difficult to find out if an Ivy artifact
     // source should be configured. At this stage, this information is lost, so we do a best effort
     // based on the file locations.
-    private static void tryHintAboutArtifactSource(TreeFormatter builder, Collection<String> locations) {
+    private void addPossibleResolution(Collection<String> locations) {
```

Comment:
❔ Is the comment above still relevant?  Probably worth converting to javadoc if so.